sdk/inc/azure/core/az_span.h

```diff
@@ -441,6 +441,46 @@ AZ_NODISCARD az_result az_span_u64toa(az_span destination, uint64_t source, az_s
 AZ_NODISCARD az_result
 az_span_dtoa(az_span destination, double source, int32_t fractional_digits, az_span* out_span);
 
+/******************************  NON-CONTIGUOUS SPAN  */
+
+/**
+ * @brief Defines a container of required and user-defined fields that provide the
+ * necessary information and parameters for the implementation of the #az_span_allocator_fn
+ * callback.
+ */
+typedef struct
+{
+  int32_t remaining_size; ///< The amount of space left within the previously provided destination,
```

Comment:
I am wondering if this should be the inverse, `int32_t bytes_used_in_previous_span`, rather than the leftover amount.

Reply:
I can't decide without seeing what customer code would look like. Can you show the 2 examples side by side for comparison?